File: TaskManager/TaskManager_210042112/TaskManager_112/TaskManagerTests/CreateIssueShouldAddIssueToRepository.cs
using TaskManager_112;

namespace TaskManagerTests
{
    public class CreateIssueShouldAddIssueToRepository
    {
        [Fact]
        public void CreateIssue_ShouldAddIssueToRepository()
        {
            var repository = new IssueRepository();
            var issue = IssueFactory.CreateIssue(1, "First Issue", "Description", Priority.Medium);
            var createIssueCommand = new CreateIssueCommand(repository, issue);
            var invoker = new CommandInvoker();
            // Act
            invoker.ExecuteCommand(createIssueCommand);
            // Assert
            var retrievedIssue = repository.GetIssue(1);
            Assert.NotNull(retrievedIssue);
            Assert.Equal("First Issue", retrievedIssue.Title);
            Assert.Equal(Priority.Medium, retrievedIssue.Priority);
            Assert.Equal(Status.Open, retrievedIssue.Status);
        }
    }
}

File: TaskManager/210042141_TaskManager/UnitTest/src/CreateIssue_ShouldAddIssueToRepository.java
import org.junit.Test;

import java.util.Date;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class CreateIssue_ShouldAddIssueToRepository {

    @Test
    public void CreateIssue_ShouldAddIssueToRepositoryTest() {
        IssueRepository repository = new IssueRepository();
        Issue issue = new Issue(1, "Issue 1", "Description", "High", "Open", new Date(), new Date(), "User1", null, null);
        Command createIssueCommand = new CreateIssueCommand(repository, issue);
        CommandInvoker invoker = new CommandInvoker();
        invoker.executeCommand(createIssueCommand);

        assertNotNull(repository.getIssue(1));
        assertEquals("High", repository.getIssue(1).getPriority());
        assertEquals("Open", repository.getIssue(1).getStatus());
    }
}
